### PR TITLE
RavenDB-13673 [NOT FOR MERGING]

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexFacetedReadOperation.cs [hunk 1]
```diff
@@ -279,9 +279,83 @@ private static void ApplyAggregation(Dictionary<string, FacetedQueryParser.Facet
             }
         }
 
+        public class NoWeightQuery : Query
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexFacetedReadOperation.cs:282]: what is this for?

File: src/Raven.Server/Documents/Queries/LuceneQueryHelper.cs [hunk 2]
```diff
@@ -137,7 +138,7 @@ public static Query Between(string fieldName, LuceneTermType termType, double fr
             return CreateRange(fieldName, fromValue, true, toValue, true);
         }
 
-        public static Query Term(string fieldName, string term, LuceneTermType type, float? boost = null, float? similarity = null, bool exact = false)
+        public static Query Term(string fieldName, string term, LuceneTermType type, float? boost = null, float? similarity = null, bool exact = false, bool lightWeight = false)
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/LuceneQueryHelper.cs:141]: I would rather make this into an instance, that has two static properties `Normal` and `Lightweight`

Comment (ayende) [reply src/Raven.Server/Documents/Queries/LuceneQueryHelper.cs:141]: So: `LuceneQueryHelper.Lightwegiht.Term(...)`

**Comment by ppekrol** [comment]
Handled here: https://github.com/ravendb/ravendb/pull/9584

**Pull request closed without merging**